src/index.ts

```diff
@@ -913,6 +934,60 @@ class ThreadPool {
 
     await Promise.all(exitEvents);
   }
+
+  async close (options : CloseOptions = kDefaultCloseOptions) {
+    this.closingUp = true;
+
+    if (options.force) {
```

Comment:
the `force` option is a bit different to `pool.destroy`

using `close` with `force: true` will wait for the running tasks to complete before terminating the workers, while `pool.destroy` will terminate all of them without waiting.

here's a small example: 

```ts
  // using destroy()

  const worker = new Piscina({
    filename: resolve(__dirname, 'fixtures/sleep.js'),
    maxThreads: 2,
  });

  setTimeout(() => {
    worker.destroy();
  }, 100);

  const result = await Promise.allSettled([
    worker.run({time: 10000, a: 1}),
    worker.run({time: 1000, a: 1}),
    worker.run({time: 1000, a: 1}),
  ])

  console.log(result); // all promises will be rejected
```

```ts
  // using close({force: true})

  const worker = new Piscina({
    filename: resolve(__dirname, 'fixtures/sleep.js'),
    maxThreads: 2,
  });

  setTimeout(() => {
    worker.close({force: true});
  }, 100);

  const x = await Promise.allSettled([
    worker.run({time: 10000, a: 1}),
    worker.run({time: 1000, a: 1}),
    worker.run({time: 1000, a: 1}),
  ])

  console.log(x); // only one will be rejected because the first two were already running
```

Reply:
And without the `force: true`?
If I'm using the API, I would expect:

1. Calling `close({ force: true })`, will shut down the pool right away,  aborting the current tasks in-flight and the remaining in the queue waiting to be processed.
2. Calling `close()`, will wait until all the tasks are being processed before shutting down the pool (maybe add a timeout to not avoid hanging threads).

One seems quite analogous to `destroy` as its basically what it does

Reply:
Without `force: true`, `close()` will wait for all the current running tasks and the enqueued tasks to complete before closing all workers. (I was also thinking about adding a timeout to the options and running destroy when the timeout expires)

I took this approach based on [this comment from Matteo](https://github.com/piscinajs/piscina/issues/139#issuecomment-883160289), but if your implementation is closer to the right path, I have no problem in closing this PR and helping with yours in any way I can :)

Reply:
Let's continue with your work, it's always nice to have new people contributing to the project 😄
Hmm, ok I see. Well, the thing is that `destroy` already does that, but it's true that it goes beyond as the current ones running are aborted. Then, wdyt if we set the following specs as a baseline for this work:

1. `Piscina#close` should stop receiving any new task, and flush the ones that are currently enqueued and in-flight (currently being processed by a worker)
2. If `Piscina#close` is called with `{ force: true }`, we abort all the `taskQueue` and `skipQueue`, but wait until the worker finishes its current tasks assigned.
3. Calling `Piscina#close`, calls `Piscina#destroy` underneath to close all the workers alive.
4. Through `new Piscina({ closeTimeout: 30000 })` we accept a timeout to use when calling close
  - If timeout is exceeded, Piscina calls `Piscina#destroy` immediately, destroying all workers and aborting all the tasks
  - The default is `30000` (we need to document this as well)

Wdyt?

Reply:
I made some updates to the code with those specs, also added a couple of tests for `timeout` cases

let me know what you think 👌